lib/ansible/modules/cloud/azure/azure_rm_keyvault.py

```diff
@@ -0,0 +1,430 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: azure_rm_keyvault
+
+version_added: "2.5"
+
+short_description: Create, delete and update key vaults.
+
+description:
+    - Creates, deletes, and updates key vaults.
+
+options:
+    resource_group:
+        description:
+            - name of resource group.
+        required: true
+    name:
+        description:
+            - name of the key vault.
+        required: true
+    state:
+        description:
+            - Assert the state of the key vault. Use 'present' to create or update and
+              'absent' to delete.
+        default: present
+        choices:
+            - absent
+            - present
+    enabled_for_deployment:
+        description:
+            - allow Virtual Machines to retrieve certificates stored as secrets from the vault.
+        default: False
+    enabled_for_disk_encryption:
+        description:
+            - allow Disk Encryption to retrieve secrets from the vault and unwrap keys.
+        default: False
+    enabled_for_template_deployment:
+        description:
+            - allow Resource Manager to retrieve secrets from the vault.
+        default: False
+    location:
+        description:
+            - Valid azure location. Defaults to location of the resource group.
+        default: resource_group location
+    sku:
+        description:
+            - SKU details.
+        default: standard
+        choices:
+            - premium
+            - standard
+    vault_uri:
+        description:
+            - the URI of the vault for performing operations on keys and secrets.
+    tenant_id:
+        description:
+            - the Azure Active Directory tenant ID that should be used for authenticating requests to the key vault.
+    object_id:
+        description:
+            - the object ID of a user, service principal or security group in the Azure Active Directory tenant for the vault.
+              the object ID must be unique for the list of access policies.
+    application_id:
+        description:
+            - application ID of the client making request on behalf of a principal.
+    permissions:
```

Comment:
This should be a lot simpler and take in differently named options than what is currently here. When having options like this it is best to keep them in a snake_case format that is simple to understand. The other point is you have predefined combination values like `Key, Secret, & Certificate Management`. This option should take in a list of values and they are combined together to form the final permissions. While some of these values may not go together (I don't know enough about access policies), the module should be able to handle that. You trade complexity in the module code for easier use by end users.

For example I would set the choices to be something like

```
default:
- key
- secret
- certificate
choices:
- key
- secret
- certificate
- sql_server
- backup
- data_lake_store
```